### PR TITLE
feat(cli): Add json output to link, unlink, allow, and allow-scrapes commands

### PR DESCRIPTION
**Pull request by adleong**
We add an `-o/--output` flag to the remaining commands which render kubernetes resources and do not yet have this flag.  The supported values for this flag are "yaml" (default) and "json".  The commands are:

* linkerd mulitcluster allow
* linkerd multicluster link
* linkerd multicluster unlink
* linkerd viz allow-scrapes

### CODE REVIEW

File: pkg/k8s/resource/resource.go [hunk 1]
```diff
@@ -112,6 +113,22 @@ func (r Kubernetes) RenderResource(w io.Writer) error {
 	return err
 }
 
+// RenderResource renders a kubernetes object as a json object
```
Comment (the-wondersmith) [inline pkg/k8s/resource/resource.go:116]:
```suggestion
// RenderResourceJSON renders a kubernetes object as a json object
```

**Comment by adleong** [comment]
Marking as draft until https://github.com/linkerd/linkerd2/pull/12641 merges so that we can reuse library functions defined there.

File: multicluster/cmd/allow.go [hunk 2]
```diff
@@ -69,16 +74,40 @@ func newAllowCommand() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			stdout.Write(buf.Bytes())
-			stdout.Write([]byte("---\n"))
 
-			return nil
+			if opts.output == "json" {
```
Comment (mateiidavid) [inline multicluster/cmd/allow.go:78]: Thought / question, would we be able to use `pkgcmd.RenderYAMLs` here?

```
pkgcmd.RenderYAMLs(&buf, os.Stdout, opts.output)
```

or would it not work? sometimes extensions have their own little logic embedded in the commands and I'm wondering if that's the case here.

File: multicluster/cmd/link.go [hunk 3]
```diff
@@ -428,7 +470,32 @@ func renderServiceMirror(values *multicluster.Values, valuesOverrides map[string
 		}
 	}
 
-	return out.Bytes(), nil
+	if format == "json" {
```
Comment (mateiidavid) [inline multicluster/cmd/link.go:473]: Same thought as above, could this make use of the library function introduced in the previous PR?

**Review by mateiidavid — APPROVED** [review]

**Pull request merged**